lib/node_modules/@stdlib/array/base/count-same-value/test/test.js

```diff
@@ -0,0 +1,89 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Complex128 = require( '@stdlib/complex/float64' );
+var Complex128Array = require( '@stdlib/array/complex128' );
+var Int32Array = require( '@stdlib/array/int32' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+var countSameValue = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof countSameValue, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function counts the number of same values (generic)', function test( t ) {
+	var expected;
+	var actual;
+	var x;
+
+	x = [ 0, 1, 0, 1, 2 ];
+	expected = 2;
+	actual = countSameValue( x, 1 );
+
+	t.strictEqual( actual, expected, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function counts the number of same values (accessors)', function test( t ) {
+	var expected;
+	var actual;
+	var x;
+
+	x = toAccessorArray( [ 0, 1, 0, 1, 2 ] );
+	expected = 2;
+	actual = countSameValue( x, 1 );
+
+	t.strictEqual( actual, expected, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function counts the number of same values (real typed array)', function test( t ) {
+	var expected;
+	var actual;
+	var x;
+
+	x = new Int32Array( [ 0, 1, 0, 1, 2 ] );
+	expected = 2;
+	actual = countSameValue( x, 1 );
+
+	t.strictEqual( actual, expected, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function counts the number of same values (complex typed array)', function test( t ) {
+	var expected;
+	var actual;
+	var x;
+
+	x = new Complex128Array( [ 0.0, 0.0, 1.0, 0.0, 3.0, 4.0, 0.0, 5.0 ] );
+	expected = 1;
+	actual = countSameValue( x, new Complex128( 3.0, 4.0 ) );
+
```

Comment:
Same comment as above. Would be useful to add test cases involving `NaN` and signed zero components.